modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ComponentModel.java

```diff
@@ -0,0 +1,266 @@
+ * but may be hold by a {@code ComponentModel}. Those attributes are retrieved by using {@code #getCustomAttributes}.
+ *
+ * @since 4.0
+ */
+public class ComponentModel
+{
+
+    private boolean root = false;
+    private ComponentIdentifier identifier;
+    private Map<String, Object> customAttributes = new HashMap<>();
+    private Map<String, String> parameters = new HashMap<>();
+    private List<ComponentModel> innerComponents = new ArrayList<>();
+    private String textContent;
+
+    //TODO  MULE-9688 Remove this attributes since should not be part of this class. This class should be immutable.
+    private BeanReference beanReference;
+    private BeanDefinition beanDefinition;
+    private Class<?> type;
+
+    /**
+     * @return the configuration identifier.
+     */
+    public ComponentIdentifier getIdentifier() {
+        return identifier;
+    }
+
+    /**
+     * @return a {@code java.util.Map} with the simple parameters of the configuration.
+     */
+    public Map<String, String> getParameters()
+    {
+        return Collections.unmodifiableMap(parameters);
+    }
+
+    /**
+     * @return a {@code java.util.List} of all the child {@code ComponentModel}s
+     */
+    public List<ComponentModel> getInnerComponents() {
+        return innerComponents;
+    }
+
+    /**
+     * @return a {@code java.util.Map} with all the custom attributes.
+     */
+    public Map<String, Object> getCustomAttributes() {
+        return Collections.unmodifiableMap(customAttributes);
+    }
+
+    /**
+     * @return true if the {@code ComponentModel} is a top level configuration element, false otherwise.
+     */
+    public boolean isRoot() {
+        return root;
+    }
+
+    /**
+     * @param beanDefinition the {@code BeanDefinition} created based on the {@code ComponentModel} values.
+     */
+    public void setBeanDefinition(BeanDefinition beanDefinition) {
+        this.beanDefinition = beanDefinition;
+    }
+
+    /**
+     * @return the {@code BeanDefinition} created based on the {@code ComponentModel} values.
+     */
+    public BeanDefinition getBeanDefinition() {
+        return beanDefinition;
+    }
+
+    /**
+     * @return the type of the object to be created when processing this {@code ComponentModel}.
+     */
+    public Class<?> getType() {
+        return type;
+    }
+
+    /**
+     * @param type the type of the object to be created when processing this {@code ComponentModel}.
+     */
+    public void setType(Class<?> type) {
+        this.type = type;
+    }
+
+    /**
+     * @return the value of the name attribute.
+     */
+    public String getNameAttribute() {
+        return parameters.get(NAME_ATTRIBUTE);
+    }
+
+    /**
+     * @return true if this {@code ComponentModel} represents a {@code org.mule.runtime.core.api.processor.MessageProcessor} scope.
+     */
+    public boolean isScope() {
+        return MessageRouter.class.isAssignableFrom(type);
```

Comment:
True. For that I've created MULE-9691 that will define a clear way to realize if the object to be build is a message processor owner or not.